### ISSUE TITLE
Filter type: "FX Comb +/-"

### ISSUE THREAD

**Issue by magnetophon** [surge-synthesizer/surge#3724]
This is the same as "FX Comb +", except the feedback slider is bipolar.

For example inn Karplus-Strong-like patches it would be nice to have both positive and negative feedback available in one filter.

**Comment by baconpaul** [comment]
This is a good idea. The reason it is hard is because right now we don't have a slider for the mix at all (just subtypes) so would have to wait until we expand params.

**Comment by magnetophon** [comment]
The resonance slider is mapped to feedback, right?
So if you make it so that the slider at minimum is -100% feedback and maximum is 100%, you are done, no?

**Comment by baconpaul** [comment]
Yes; but that's a different feedback than the +/- 50 100 feedback, oddly.

That said: making resonance go -1/1 for all the combs is a good idea 

the algorithms is basically

```
d = in + reso * delayval
d = softclip(d)
delay_buffer[writepoint] = d
out = mix * d + (1-mix) * in
```

where 'mix' is -100, -50, 50, or 100 depending on the +/- and the subtype

So really 'bipolar resonance' would work for each of comb +/- and their associated subtypes

and what you really want is a third slider which is that mix as opposed to discrete subtypes.

**Comment by mkruselj** [comment]
But wait the negative mix would basically give you the negative comb, so there's no need to have bipolar resonance if you have bipolar mix. No?

**Comment by baconpaul** [comment]
so the resonance controls what gets written to the delay buffer
the mix controls what gets read out of it
separate things

**Comment by baconpaul** [comment]
> So in Surge XT, Comb+ and Comb- become a single filter type with a single subtype, because bipolar Mix slider covers all those cases.

yes. bipolar resonance and bipolar mix as two sliders is the best solution

**Comment by j5v** [comment]
The only difference in behaviour between old and new patches, would be the clipping* of out-of-range feedback values (from modulations). 
*This carries the assumption that the values are clipped.

**Comment by magnetophon** [comment]
Any news on this?

**Comment by mkruselj** [comment]
Not gonna be in XT1.0 (see attached milestone) 🙂

**Comment by mkruselj** [comment]
This has been implemented in Shortcircuit and factored out to sst-filters. I consider this issue resolved ahead of time - when we bring in the comb filter code in XT2.0 it will already have this feature.

**Comment by Andreya-Autumn** [comment]
The Shortcircuit one uses SIMD to get both at once in a way that wouldn't work with the Surge voice strategy. But we can work that out when we get there I guess.
